src/copy.rs

```diff
@@ -149,42 +150,68 @@ where
 {
     let (mut rd, mut wd) = downstream.split_into_buffered_reader();
     let (mut ru, mut wu) = upstream.split_into_buffered_reader();
-
     let downstream_to_upstream = async {
-        let res = copy_buf(&mut rd, &mut wu, stats, false).await;
+        let report = |e: io::Error| {
+            SendError(Box::new(match e.kind() {
+                io::ErrorKind::NotConnected => BackendDisconnected,
```

Comment:
`downstream_to_upstream` matches errors `NotConnected` and `WriteZero` to `BackendDisconnected` whereas `upstream_to_downstream` does the opposite and matches `NotConnected` and `WriteZero` to `ClientDisconnected`

Can you provide context for why it's different?

Reply:
Yes, these are opposite ends of the connection. So they indicate opposite sides. its bidirectional

Reply:
Ah ok that makes sense, thanks for the clarification!